src/functions-and-arrays.js

```diff
@@ -26,13 +52,40 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
-
+function averageNumbers(numbersAvg) {
+
+  let sumNumbers = 0
+  
+  for (let i=0; i < numbersAvg.length; i++ ){
+   sumNumbers += numbersAvg[i] ;
+  }
+  let  averageNumber = sumNumbers / numbersAvg.length;
+  //if(numbersAvg === []){
+  //return 0
+//}
+  return averageNumber
+}
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+  let sumOfArr = 0
+  let averageLength = 0
+  let fullLengthWords = []
+  if(wordsArr.length >0){
+    for(let i= 0; i<wordsArr.length; i++){
+      fullLengthWords.push(wordsArr[i].length)
+    }      
+   }
+   for(let j=0; j<fullLengthWords.length;j++){
+   sumOfArr += fullLengthWords[j];
+   averageLength = sumOfArr / wordsArr.length
+   }
+  //else{
+    //return null
+  //}
+  return averageLength
+}
```

Comment:
Also die Aufgaben scheinen nicht das Problem gewesen zu sein, sondern vielleicht eher die Information aus dem Test zu verstehen? Das war bisschen tricky insofern, dass manches nicht in der Aufgabe selber erwähnt wurde, sondern lediglich im Test um euch sozusagen bisschen dahingehend zu trainieren, dass ihr mit Tests umgehen könnt